payment-service/src/main/java/by/sergo/paymentservice/service/impl/CreditCardServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+
+        if (passengerCreditCard.getBalance().compareTo(payment.getSum()) >= 0) {
+            passengerCreditCard.setBalance(passengerCreditCard.getBalance().subtract(payment.getSum()));
+        } else {
+            throw new BadRequestException(ExceptionMessageUtil.getWithdrawalExceptionMessage("Credit card", "passengerId", payment.getPassengerId().toString(), passengerCreditCard.getBalance().toString()));
+        }
+
+        var savedCreditCard = creditCardRepository.save(passengerCreditCard);
+        driverAccount.setBalance(driverAccount.getBalance().add(payment.getSum()));
+        accountRepository.save(driverAccount);
+        var transaction = TransactionStore.builder()
+                .accountNumber(driverAccount.getAccountNumber())
+                .creditCardNumber(savedCreditCard.getCreditCardNumber())
+                .value(payment.getSum())
+                .operationDate(LocalDateTime.now())
+                .operation(PAYMENT)
+                .build();
+        transactionStoreRepository.save(transaction);
+        return creditCardMapper.mapToDto(savedCreditCard);
+    }
+
+    private CreditCard getByIdOrElseThrow(Long id) {
+        return creditCardRepository.findById(id)
+                .orElseThrow(() -> new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Credit card", "id", id)));
+    }
+
+    private CreditCard getPassengerCreditCard(Long passengerId) {
+        return creditCardRepository.findByUserIdAndUserType(passengerId, UserType.PASSENGER)
+                .orElseThrow(() -> new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Credit card with type passenger", "userId", passengerId)));
+    }
+
+    private void checkCardIsUniqueForUpdate(CreditCardCreateUpdate dto, Long id) {
+        if (!creditCardRepository.existsById(id)){
+            throw new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Credit card", "id", id));
+        }
+        checkUniqueCreditCardAndExpirationDate(dto);
+
+        if (creditCardRepository.existsByUserIdAndUserType(dto.getUserId(), UserType.valueOf(dto.getUserType()))){
+            throw new BadRequestException(ExceptionMessageUtil.getAlreadyExistMessage("Credit Card", "userId", dto.getUserId(), "userType", dto.getUserType()));
+        }
+    }
+
+    private void checkUniqueCreditCardAndExpirationDate(CreditCardCreateUpdate dto) {
+        if (creditCardRepository.existsByCreditCardNumber(dto.getCreditCardNumber())) {
+            throw new BadRequestException(ExceptionMessageUtil.getAlreadyExistMessage("Credit Card", "card number", dto.getCreditCardNumber()));
+        } else {
+            if (dto.getExpDate().isBefore(LocalDate.now())) {
+                throw new BadRequestException(ExceptionMessageUtil.getExpirationCardExceptionMessage("Credit card", "expiration date", dto.getExpDate().toString()));
+            }
```

Comment:
Можно сделать без else
```suggestion
            if (creditCardRepository.existsByCreditCardNumber(dto.getCreditCardNumber())) {
            throw new BadRequestException(ExceptionMessageUtil.getAlreadyExistMessage("Credit Card", "card number", dto.getCreditCardNumber()));
        }
        if (dto.getExpDate().isBefore(LocalDate.now())) {
            throw new BadRequestException(ExceptionMessageUtil.getExpirationCardExceptionMessage("Credit card", "expiration date", dto.getExpDate().toString()));
        }
```